R/pkg/R/sparkR.R

```diff
@@ -365,6 +365,10 @@ sparkR.session <- function(
     }
     overrideEnvs(sparkConfigMap, paramMap)
   }
+  if (nzchar(master)) {
+    assign("spark.master", master, envir = sparkConfigMap)
```

Comment:
Yeah this is a tough one - I think preferring `master=` makes as it follows from a convention that the explicit arguments take precedence over more implicit ones passed as a list or read from a file. Also I think mimicing what `bin/spark-shell` does is probably good for consistency.

The other option is to just throw an error and let the user fix this ?